### PR TITLE
Move getNextPod to closure

### LINKED ISSUE #72546: [scheduler cleanup phase 🐼]: Change `getNextPod` from method to function
**PLEASE COORDINATE WITH SIG-SCHEDULING BEFORE WORKING ON THIS ISSUE.** We want to avoid the situation where two people accidentally start work on the same fix.

Main tracking issue: https://github.com/kubernetes/kubernetes/issues/68951

The `getNextPod` method of `configFactory` is a simple wrapper around `SchedulingQueue.Pop`. It should be converted into a simple closure instead of being a method. i.e. we should have,

```golang

func MakeNextPodFunc(queue SchedulingQueue) func() *v1.Pod {...}

```

(code is in `pkg/scheduler/factory/factory.go`)

/sig scheduling

/kind cleanup

/help

### PR DESCRIPTION
**Pull request by danielqsj**
**What type of PR is this?**

/sig scheduling
/kind cleanup

**What this PR does / why we need it**:
The `getNextPod` method of `configFactory` is a simple wrapper around `SchedulingQueue.Pop`. It should be converted into a simple closure instead of being a method.

**Which issue(s) this PR fixes**:

Fixes #72546

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by danielqsj** [comment]
cc @misterikkit

File: pkg/scheduler/internal/queue/scheduling_queue.go [hunk 1]
```diff
@@ -808,3 +808,15 @@ func newNominatedPodMap() *nominatedPodMap {
 		nominatedPodToNode: make(map[ktypes.UID]string),
 	}
 }
+
```
Comment (wgliang) [inline pkg/scheduler/internal/queue/scheduling_queue.go]: Need `MakeNextPodFunc`  function comment.

**Review by neolit123 — APPROVED** [review]
/priority backlog

File: pkg/scheduler/internal/queue/scheduling_queue.go (hunk 1 above)
Comment (danielqsj) [reply pkg/scheduler/internal/queue/scheduling_queue.go]: @wgliang  comment added.

**Comment by wgliang** [comment]
Please squash your commits. Thanks.

**Review by misterikkit — COMMENTED** [review]
/lgtm

Thanks for the quick work on this one.

Optionally, could you squash the two commits?

**Review by ravisantoshgudimetla — APPROVED** [review]
/lgtm

Thanks for the PR @danielqsj

**Pull request merged**